driver-core/src/test/unit/com/mongodb/MongoClientSettingsSpecification.groovy

```diff
@@ -401,26 +403,63 @@ class MongoClientSettingsSpecification extends Specification {
         expect expectedSettings, isTheSameAs(settingsWithDefaultConnectionStringApplied)
     }
 
-    def 'should use the socket settings connectionTime out for the heartbeat settings'() {
+    def 'should use the socket settings connectionTimeout for the heartbeat settings'() {
         when:
-        def settings = MongoClientSettings.builder().applyToSocketSettings(new Block<SocketSettings.Builder>() {
-            @Override
-            void apply(final SocketSettings.Builder builder) {
-                builder.connectTimeout(42, TimeUnit.SECONDS).readTimeout(60, TimeUnit.SECONDS).receiveBufferSize(22).sendBufferSize(10)
-            }
-        }).build()
-        def expected = SocketSettings.builder().connectTimeout(42, TimeUnit.SECONDS).readTimeout(42, TimeUnit.SECONDS).build()
+        def settings = MongoClientSettings.builder().applyToSocketSettings { SocketSettings.Builder builder ->
+            builder.connectTimeout(42, TimeUnit.SECONDS).readTimeout(60, TimeUnit.SECONDS)
+                    .receiveBufferSize(22).sendBufferSize(10)
```

Comment:
Nit: remove settings not tested: `.receiveBufferSize(22).sendBufferSize(10)`

Reply:
I think I did this purposely to show that we use the default values for those and don't take them from the socket settings, so I would just assume to leave it as is.

Reply:
OK - sgtm